Samples/WindowsML/Resources/SqueezeNetModel.targets

```diff
@@ -0,0 +1,72 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <!-- Define properties for model download -->
+  <PropertyGroup>
+    <!-- Model URL -->
+    <SqueezeNetModelUrl>https://github.com/onnx/models/raw/b1eeaa1ac722dcc1cd1a8284bde34393dab61c3d/validated/vision/classification/squeezenet/model/squeezenet1.1-7.onnx</SqueezeNetModelUrl>
+    <SqueezeNetLabelsUrl>https://github.com/microsoft/Windows-Machine-Learning/raw/02b586811c8beb1ae2208c8605393267051257ae/Samples/SqueezeNetObjectDetection/Desktop/cpp/Labels.txt</SqueezeNetLabelsUrl>
+    <SqueezeNetImageUrl>https://github.com/microsoft/Windows-Machine-Learning/raw/638cd40abc408e7049219e5d7b4fe164c304f2a2/SharedContent/media/kitten_224.png</SqueezeNetImageUrl>
+    
+    <!-- Use a shared location under Resources/obj for all projects -->
+    <SqueezeNetModelTempDir>$(MSBuildThisFileDirectory)obj\SqueezeNetModel</SqueezeNetModelTempDir>
+    
+    <!-- Model file paths -->
+    <SqueezeNetModelFile>$(SqueezeNetModelTempDir)\SqueezeNet.onnx</SqueezeNetModelFile>
+    <SqueezeNetLabelsFile>$(SqueezeNetModelTempDir)\SqueezeNet.Labels.txt</SqueezeNetLabelsFile>
+    <SqueezeNetImageFile>$(SqueezeNetModelTempDir)\image.png</SqueezeNetImageFile>
+  </PropertyGroup>
+
+  <!-- Create temp directory for downloaded model -->
+  <Target Name="CreateSqueezeNetModelDirectory" BeforeTargets="Build">
+    <MakeDir Directories="$(SqueezeNetModelTempDir)" Condition="!Exists('$(SqueezeNetModelTempDir)')" />
+  </Target>
+
+  <!-- Download SqueezeNet model if it doesn't exist -->
+  <Target Name="DownloadSqueezeNetModel" BeforeTargets="Build" DependsOnTargets="CreateSqueezeNetModelDirectory">
+    <Message Text="Checking for SqueezeNet model in: $(SqueezeNetModelTempDir)" Importance="high" />
+    
+    <!-- Download model file if it doesn't exist -->
+    <DownloadFile 
+      SourceUrl="$(SqueezeNetModelUrl)" 
+      DestinationFolder="$(SqueezeNetModelTempDir)" 
+      DestinationFileName="SqueezeNet.onnx"
+      Condition="!Exists('$(SqueezeNetModelFile)')"
+      ContinueOnError="true" />
+    <Warning Text="SqueezeNet model download may have been interrupted by antivirus scanning or network issues. The file may become available shortly as antivirus completes its scan."
```

Comment:
>Warning [](http://example.com/codeflow?start=5&length=7)

Looking at the task documentation, it sounds like it has retry and delay functionality which could be considered to further harden against this. [https://learn.microsoft.com/en-us/visualstudio/msbuild/downloadfile-task?view=vs-2022](https://learn.microsoft.com/en-us/visualstudio/msbuild/downloadfile-task?view=vs-2022)